pandas/core/strings.py

```diff
@@ -2325,12 +2206,127 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
             res = Series(res, index=data.index, name=self._orig.name)
         return res
 
-    @copy(str_split)
+    _shared_docs['str_split'] = ("""
+    Split strings around given separator/delimiter.
+
+    Splits the string in the Series/Index from the %(side)s,
+    at the specified delimiter string. Equivalent to :meth:`str.%(method)s`.
+
+    Parameters
+    ----------
+    pat : str, optional
+        String or regular expression to split on.
+        If not specified, split on whitespace.
+    n : int, default -1 (all)
+        Limit number of splits in output.
+        ``None``, 0 and -1 will be interpreted as return all splits.
+    expand : bool, default False
+        Expand the splitted strings into separate columns.
+
+        * If ``True``, return DataFrame/MultiIndex expanding dimensionality.
+        * If ``False``, return Series/Index, containing lists of strings.
+
+    Returns
+    -------
+    Series, Index, DataFrame or MultiIndex
+        Type matches caller unless ``expand=True`` (see Notes).
+
+    Notes
+    -----
+    The handling of the `n` keyword depends on the number of found splits:
+
+    - If found splits > `n`,  make first `n` splits only
+    - If found splits <= `n`, make all splits
+    - If for a certain row the number of found splits < `n`,
+      append `None` for padding up to `n` if ``expand=True``
+
+    If using ``expand=True``, Series and Index callers return DataFrame and
+    MultiIndex objects, respectively.
+
+    See Also
+    --------
+     Series.str.split : Split strings around given separator/delimiter.
+     Series.str.rsplit : Splits string around given separator/delimiter, starting from 
+     the right.
+     Series.str.join : Join lists contained as elements in the Series/Index 
+     with passed delimiter. 
+     str.split : Standard library version for split.
+     str.rsplit : Standard library version for rsplit.
+
+    Examples
+    --------
+    >>> s = pd.Series(["this is a regular sentence", "this,is,comma,separated,text", np.nan])
+
+    By default, split and rsplit will return an object of the same size
```

Comment:
I'd start by the simplest possible example, with all the arguments with default values: `>>> s.str.split()`. I think for many users that's the only example they'll care about when checking this doc.

Reply:
I was thinking about that - but then there would be a bit redundancy further down. I could rearrange it look something like this - what do you think ? 

### Default arguments
s.str.split()
s.str.rsplit()

## semicolon separator or path separator
s.str.split(";")
s.str.rsplit(";")

### number of splits
s.str.split(n=2)
s.str.rsplit(n=2)

### with expand argument
s.str.split(n=2, expand = True)
s.str.rsplit(n= 2, expand = True)